### PR TITLE
[Serializer] Docs for the @MaxDepth annotation

### PR DESCRIPTION
**Pull request by dunglas**
Docs for symfony/symfony#17113.

### CODE REVIEW

File: components/serializer.rst
Comment (snoek09) [inline components/serializer.rst]: indent php code block

File: components/serializer.rst
Comment (snoek09) [inline components/serializer.rst]: Add missing `.. code-block:: php-annotations`.

Comment (wouterj) [inline components/serializer.rst]: Actually, just `.. code-block:: php`

**Review by javiereguiluz — APPROVED** [review]
👍 a useful feature explained in a clear and concise way. Thanks @dunglas.

**Comment by lyrixx** [comment]
@javiereguiluz Hello. I was playing with the serializer and I did not find the doc on symfony.com. You added your :+1: 15 days ago. I think we can merge this one to be able to deploy this new documentation? (It's available since symfony 3.1 ...)

**Comment by javiereguiluz** [comment]
@lyrixx you are so right!! But before merging PRs in Symfony Docs, we need two 👍 from maintainers. So we need that @weaverryan or @xabbuh or @wouterj vote on this too. Thanks!

**Comment by digilist** [comment]
I have a question regarding the max depth. I am not sure if this is desired behavior or a bug. If its the former, we should document this. If its the latter, we should fix the bug (I think it's a bug, but I wanted to ask you before opening an issue).

Consider some code that extends your example code with more levels (4, 5, etc.). By keeping a max depth of 2, the result is the following:


        'foo' => 'level1',
        'child' =>
            array (
                'foo' => 'level2',
                'child' =>
                    array (
                        'child' =>
                            array (
                                'child' =>
                                    array (
                                        'child' =>
                                            array (
                                                'child' => NULL,
                                            ),
                                    ),
                            ),
                    ),
            ),
    )

As you can see, it considers all child nodes and does not stop after the configured max depths. Only the data of the object is ignored (attribute foo).

**Comment by xabbuh** [comment]
Thank you Kévin.

**Comment by xabbuh** [comment]
@digilist I think best is if you could open an issue on the code repository containing the necessary code to reproduce your issue.

**Pull request merged**
